dbcon/mysql/ha_mcs_impl.cpp

```diff
@@ -3086,6 +3086,10 @@ void ha_mcs_impl_start_bulk_insert(ha_rows rows, TABLE* table, bool is_cache_ins
             //@bug 6122 Check how many columns have not null constraint. columnn with not null constraint will not show up in header.
             unsigned int numberNotNull = 0;
 
+            // There's a chance that a previously aborted query left garbage in ci->columnTypes
```

Comment:
IMHO This change is outside the scope of the patch and musn't get into this patch. Moreover the clean-up made in the middle of an operation looks like a side-effect nobody knows about so we better put it where all clean-up happens, e.g. impl_external_lock().

Reply:
I disagree. This is something that belongs here. There is a clear() at the tail of the function, but errors may cause early exit, bypassing the clear(). It is common to initialize variables before use. In this case, it's a good safe way to do it.
As for scope, that can be debated, but uninitialized ci->columnTypes caused a bit of pain while testing this patch. Thus this change. To just remove it as not in the scope of the problem will cause unnecessary headache. We'd end up putting it back in at a later time. Or perhaps find every branch that exits the function and add cleanup code? Nah.

Reply:
No query passes impl_external_lock() and adding uncorrelated side effects makes it hard to debug/maintain the code IMHO. That is why I am strongly against putting it in the middle of  ha_mcs_impl_start_bulk_insert

Reply:
I removed the offensive code for now and opened MCOL-4968 to address the issue.